webapp/channels/src/components/advanced_text_editor/use_post_box_indicator.tsx

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {DateTime} from 'luxon';
+import {useState, useEffect, useMemo} from 'react';
+import {useSelector} from 'react-redux';
+
+import {getDirectChannel} from 'mattermost-redux/selectors/entities/channels';
+import {isScheduledPostsEnabled} from 'mattermost-redux/selectors/entities/scheduled_posts';
+import {getTimezoneForUserProfile, getCurrentTimezone} from 'mattermost-redux/selectors/entities/timezone';
+import {getStatusForUserId, getUser, makeGetDisplayName} from 'mattermost-redux/selectors/entities/users';
+
+import Constants, {UserStatuses} from 'utils/constants';
+
+import type {GlobalState} from 'types/store';
+
+const DEFAULT_TIMEZONE = {
+    useAutomaticTimezone: true,
+    automaticTimezone: '',
+    manualTimezone: '',
+};
+
+const MINUTE = 1000 * 60;
+
+function useTimePostBoxIndicator(channelId: string) {
+    const getDisplayName = useMemo(makeGetDisplayName, []);
+
+    // Get the teammate ID from DM associated withchannel ID
+    const teammateId = useSelector((state: GlobalState) => getDirectChannel(state, channelId)?.teammate_id || '');
+    const teammateDisplayName = useSelector((state: GlobalState) => (teammateId ? getDisplayName(state, teammateId) : ''));
+
+    // check if the teammate is in DND status
+    const showDndWarning = useSelector((state: GlobalState) =>
+        (teammateId ? getStatusForUserId(state, teammateId) === UserStatuses.DND : false),
+    );
+
+    // Determine if the current channel is a DM and the teammate is NOT in DND
+    const isDM = useSelector(
```

Comment:
It looks a bit weird that `isDM` depends on `showDndWarning`, I'd think the opposite, `showDndWarning` depending on `isDM` like how it was in the `PostBoxIndicator` component. Can you update this so the semantics are correct?

Reply:
nice catch! done